### ISSUE TITLE
Offline installation of Pygrampc

### ISSUE THREAD

**Issue by karthyyy** [grampc/pygrampc#3]
Hi,

  I am trying to install "pygrampc" offline (PC not connected to internet). Usually I download all the required packages from online PC and then transfer to offline PC and install it. But in the repo there are no dependency requirements mentioned. So I cloned the repo, went to the folder and ran "pip install -e ." . Its returning following error.

"WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x000001A7C49BAF40>, 'Connection to pypi.org timed out. (connect timeout=15)')': /simple/scikit-build-core/
ERROR: Could not find a version that satisfies the requirement scikit-build-core>=0.3.3 (from versions: none)
ERROR: No matching distribution found for scikit-build-core>=0.3.3"

  So I installed scikit-build = 0.17.6 and scikit-build-core = 0.8.1. But its still showing the same error.

  I tried installing the same in an online PC under a different env and its working fine. 

   Can you please help with this.

Thanks and Regards
    Karthik

**Comment by ThoreWietzke** [comment]
In python packaging invoking "pip install ." calls the build frontend, which then finds the required build backend, in this case scikit-build-core. The build is done in an isolated environment, so pip tries to download the dependencies and ignores your already installed packages. 

You can try 
`pip install -e --no-build-isolation .`
to disable the build isolation.

**Comment by karthyyy** [comment]
Hi,
 
  Sorry, I missed the earlier reply.

  I executed the command `pip install -e --no-build-isolation .` which returned the following error

**ERROR: --no-build-isolation is not a valid editable requirement. It should either be a path to a local project or a VCS URL (beginning with bzr+http,**

  I changed the command to `pip install -e . --no-build-isolation` which initiated building the packages and requested scikit-build-core, pyproject_metadata and pathspec, which I installed in offline PC. Even after installing all these, running `pip install -e . --no-build-isolation` is returning following unknown error.

 **CMake Error at CMakeLists.txt:5 (find_package):
        Could not find a package configuration file provided by "pybind11" with any
        of the following names:

          pybind11Config.cmake
          pybind11-config.cmake

        Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
        "pybind11_DIR" to a directory containing one of the above files.  If
        "pybind11" provides a separate development package or SDK, be sure it has
        been installed.


      -- Configuring incomplete, errors occurred!

      *** CMake configuration failed
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building editable for pygrampc**

    Please help with the same.

Thanks and Regards
   Karthik R
